File: selection-sort/SelectionSort.swift
import Foundation

class SelectionSort {
    func selectionSort(_ array: inout [Int]) -> [Int] {
        for i in 0 ..< array.count {
            var j = i + 1
            var smallestIndex = i
            while(j < array.count) {
                if(array[smallestIndex] > array[j]) { smallestIndex = j }
                j += 1  
            }
            array.swapAt(i, smallestIndex)
        }
        return array
    }
}

File: selection-sort/SelectionSort.java
public class SelectionSort {
    public int[] selectionSort(int[] array) {
        for(int i = 0; i < array.length; i++) {
            int j = i + 1;
            int smallestIndex = i;
            while(j < array.length) {
                if(array[smallestIndex] > array[j]) {
                    smallestIndex = j;
                }
                j++;
            }
            int temp = array[smallestIndex];
            array[smallestIndex] = array[i];
            array[i] = temp;
        }
        return array;
    }
}
